File: packages/sdui/src/main/java/nl/q42/server/middleware/AppLocaleMiddleware.java
package nl.q42.server.middleware;

import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import lombok.extern.slf4j.Slf4j;
import nl.q42.core.Locale;
import org.springframework.stereotype.Component;
import org.springframework.web.servlet.HandlerInterceptor;

@Component
@Slf4j(topic = "App Locale Middleware")
public class AppLocaleMiddleware implements HandlerInterceptor
{

  static final Locale DEFAULT_LOCALE = Locale.NL_NL;

  @Override
  public boolean preHandle(
      HttpServletRequest request,
      HttpServletResponse response,
      Object handler
  )
  {
    var appLocale = Locale.from(
        request.getHeader(MiddlewareConfiguration.HEADER_APP_LOCALE),
        DEFAULT_LOCALE
    );

    request.setAttribute(MiddlewareConfiguration.ATTRIB_LOCALE, appLocale);

    return true;
  }
}

File: packages/sdui/src/main/kotlin/nl/q42/server/middleware/AppLocaleMiddleware.kt
package nl.q42.server.middleware

import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse
import nl.q42.common.RequestHeader
import nl.q42.common.core.Locale
import org.springframework.stereotype.Component
import org.springframework.web.servlet.HandlerInterceptor

@Component
class AppLocaleMiddleware : HandlerInterceptor {
    override fun preHandle(
        request: HttpServletRequest,
        response: HttpServletResponse,
        handler: Any
    ): Boolean {
        val appLocale: Locale = Locale.from(
            request.getHeader(RequestHeader.HEADER_APP_LOCALE),
            DEFAULT_LOCALE
        )

        request.setAttribute(RequestHeader.ATTRIB_LOCALE, appLocale)

        return true
    }

    companion object {
        val DEFAULT_LOCALE: Locale = Locale.NL_NL
    }
}
